swift/ql/test/query-tests/Security/CWE-611/testXXE.swift

```diff
@@ -0,0 +1,58 @@
+// --- stubs ---
+
+class Data {
+    init<S>(_ elements: S) {}
+}
+
+struct URL {
+	init?(string: String) {}
+}
+
+class InputStream {
+    init(data: Data) {}
+}
+
+extension String {
+	init(contentsOf: URL) {
+        let data = ""
+        self.init(data)
+    }
+}
+
+class XMLParser {
+    var shouldResolveExternalEntities: Bool { get { return false } set {} }
+    init?(contentsOf: URL) {}
+    init(data: Data) {}
+    init(stream: InputStream) {}
+}
+
+// --- tests ---
+
+func testData() {
+    let remoteString = String(contentsOf: URL(string: "http://example.com/")!)
+    let remoteData = Data(remoteString)
+    let parser = XMLParser(data: remoteData) // $ hasXXE=32
+    parser.shouldResolveExternalEntities = true
+}
```

Comment:
I notice that we don't care where the parser is used after its set up incorrectly.  I think its fair to assume that it will be in the case above, why else would it be created like that?  But for libraries where the default configuration is unsafe there's a plausible edge case:
```
let parser = UnsafeByDefaultParser(data: remoteData)
parser.parse(...) // BAD
parser.shouldResolveExternalEntities = false // makes it safe, but the above is still BAD
```
How concerned are you about this case?  I believe the CPP version of the query does account for it, I'm not entirely sure if its worth the effort though.

Reply:
I expected to model other libraries differently (since the `shouldResolveExternalEntities` attribute will probably have a different name anyway). But in any case, the current implementation requires `true` to be set to `shouldResolveExternalEntities`, so setting it to `false` shouldn't flag the parser as vulnerable.

To be 100% sure, I added a couple of test cases in e1aa75a.

Reply:
> I notice that we don't care where the parser is used after its set up incorrectly.

We could require a [`parse()`](https://developer.apple.com/documentation/foundation/xmlparser/1411778-parse) call with even more local data flow, but my concern is that we'd miss cases where local flow isn't enough to track the vulnerable parser to the location where `parse()` is called. And, as you said, normally a parser wouldn't be initialized to then not be used (and still, if that was the case, it would be a vulnerability waiting to happen which should be fixed IMHO).